### ISSUE TITLE
NVDA browse mode can't catch changed aria-current state if DOM is not refreshed in Chrome.

### ISSUE THREAD

**Issue by khsbory** [nvaccess/nvda#11761]
Hello, my name is Hyongsop Kim from Korea and NVDA user.
NVDA support aria-current attribute in HTML very well.
As you know, aria-current attribute is used which element is current page or step and etc among the items.
By the way, in Chrome browser, there is a NVDA bug.
Let’s suppose that there are for links and first has aria-current set to true.
If I change link by pressing enter key on a second or third link, that link will take aria-current attribute set to true and rests are false.
By the way, if I press the enter key on a link and HTML DOM is refreshed, NVDA browse mode recognize changed aria-current attribute.
But when I press the button with aria-current property set to false, it is changed to true and if HTML DOM is not refreshed, NVDA browse mode does not recognize the changed aria-current property.
You can play [this bug report video](https://khsbory.github.io/nvda-bug-report-about-ariacurrent-with-chrome.mp4).
I demonstrated this issue with video recording.
### Steps to reproduce:
1. Go to [this site](https://khsbory.github.io/video%20player%20with%20aria-live%20script.html) with chrome.
2. Go to the main landmark section.
In main landmark, there are some changing video state buttons.
3. Press enter the buttons.
Whenever you enter the button, the button with aria-current attribute will set to true, others will set to false.
### Actual behavior:
NVDA browse mode doesn't catch the aria-current state.
### Expected behavior:
NVDA browse mode should catch the changed aria-current state.

**Comment by feerrenrut** [comment]
Hi @khsbory thank you for raising this issue. 
I can reproduce this with NVDA 2020.3 and Chrome canary `88.0.4294.0 `
I can not reproduce the issue with Firefox `81.0.2 (64-bit)`

As suggested by the results of refreshing the NVDA virtual buffer, I can confirm that the appropriate element has aria-current set with the Chrome accessibility tool.

I suspect that Chrome is not sending an event, or we are missing it somehow. Bear in mind, NVDA uses almost exactly the code for Chrome and Firefox.

**Comment by feerrenrut** [comment]
@ObjectInSpace are you aware of this issue?

**Comment by khsbory** [comment]
Thank you for your checking.
Once I turn off browse mde and press the button, aria-current attribute is changed well.
So I think that NVDA browse mode can't catch directly changed attribute.
Thank you.

**Comment by feerrenrut** [comment]
Just to simplify debugging this I have created a simpler example: https://cdpn.io/reefturner/debug/yLJOavW/bYrdyXBddaZA

**Comment by ObjectInSpace** [comment]
Not aware of this AFAIK. I got a 403 when trying to load the codepen example.

**Comment by khsbory** [comment]
@ObjectInSpace You can test this issue on [this page](file:///D:/project/khsbory.github.io/video%20player%20with%20aria-live%20script.html).

**Comment by feerrenrut** [comment]
Hmm, I probably should have included a direct link to the pen: https://codepen.io/reefturner/pen/yLJOavW
The debug view is typically easier for testing with NVDA.
Here is the "full" view which might remain available. https://codepen.io/reefturner/full/yLJOavW

**Comment by ObjectInSpace** [comment]
Thanks much! Filed https://bugs.chromium.org/p/chromium/issues/detail?id=1141139 for this issue.

**Comment by feerrenrut** [comment]
We expect `IA2_EVENT_OBJECT_ATTRIBUTE_CHANGED` for this, which is what Firefox does. Chrome doesn't send that event, instead it is sending `IA2_EVENT_TEXT_ATTRIBUTE_CHANGED` for this example. We ignore that event in our virtual buffer code currently, and I'm not sure why yet.

I confirmed this using AccEvent, filtering only to Chrome Hwnd (and descendants) after starting Chrome with `--force-renderer-accessibility` (necessary if you don't have a screen reader running). I used spy++ to get the Chrome Hwnd. Note that `IA2_EVENT_TEXT_ATTRIBUTE_CHANGED` is reported as "unknown" by AccEvent, the event value is `0x11A`, these enums are listed in `include\ia2\api\AccessibleEventID.idl`, but it is easier to look up a value via the NVDA python console (<kbd>NVDA+control+z</kbd>):

``` python
>>> import comInterfaces.IAccessible2Lib
>>> hex(comInterfaces.IAccessible2Lib.IA2_EVENT_TEXT_ATTRIBUTE_CHANGED)
'0x11a'
>>> import IAccessibleHandler
>>> IAccessibleHandler.getWinEventName(0x11a)
'IA2_EVENT_TEXT_ATTRIBUTE_CHANGED'
```

If NVDA watches for `IA2_EVENT_TEXT_ATTRIBUTE_CHANGED` as well, the issue is resolved.

**Comment by khsbory** [comment]
Hello.
I found that this issue is a little fixed with NVDA latest alpha.
Thank you for your effort to fix it.
As you said, chrome is firing IA2_EVENT_TEXT_ATTRIBUTE_CHANGED event, not IA2_EVENT_OBJECT_ATTRIBUTE_CHANGED event.
So even if aria-current attribute is set to true, NVDA doesn't read the changed attribute right away once user press the button.
I think chrome should fix the event property that firefox is using in order to read changed attribute right away.
However once I change focus and back to changed object, NVDA browse mode read changed aria-current attribute.
Thank you so much again.

**Comment by Adriani90** [comment]
Cc: @ObjectInSpace

**Comment by Adriani90** [comment]
This is fixed in Chrome 111 with NVDA 2023.1 RC1. NVDA reports the state change when pressing enter. Closing as works for me.
